src/components/picker/PickerItem.js

```diff
@@ -16,153 +16,138 @@ import Text from '../text';
  * @extendslink: docs/TouchableOpacity
  * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/PickerScreen.js
  */
-class PickerItem extends BaseComponent {
-  static displayName = 'Picker.Item';
-
-  static propTypes = {
-    /**
-     * Item's value
-     */
-    value: PropTypes.oneOfType([PropTypes.object, PropTypes.string, PropTypes.number]),
-    /**
-     * Item's label
-     */
-    label: PropTypes.string,
-    /**
-     * Custom function for the item label (e.g (value) => customLabel)
-     */
-    getItemLabel: PropTypes.func,
-    /**
-     * DEPRECATE: Function to return the value out of the item value prop when value is custom shaped.
-     */
-    getItemValue: PropTypes.func,
-    /**
-     * Is the item selected
-     */
-    isSelected: PropTypes.bool,
-    /**
-     * Pass to change the selected icon
-     */
-    selectedIcon: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
-    /**
-     * Pass to change the selected icon color
-     */
-    selectedIconColor: PropTypes.string,
-    /**
-     * Is the item disabled
-     */
-    disabled: PropTypes.bool,
-    /**
-     * Render custom item
-     */
-    renderItem: PropTypes.elementType,
-    /**
-     * Callback for onPress action
-     */
-    onPress: PropTypes.func,
-    /**
-     * Callback for onLayout event
-     */
-    onSelectedLayout: PropTypes.func
-  };
-
-  
-  constructor(props) {
-    super(props);
-
-    if (_.isPlainObject(props.value)) {
+const PickerItem = props => {
+  const {
+    renderItem,
+    value,
+    label,
+    onPress,
+    disabled,
+    isSelected,
+    selectedIcon = Assets.icons.check,
+    selectedIconColor,
+    testID
+  } = props;
+
+  useEffect(() => {
+    if (_.isPlainObject(value)) {
       console.warn('UILib Picker.Item will stop supporting passing object as value & label (e.g {value, label}) in the next major version. Please pass separate label and value props');
     }
-  }
-  
+  }, []);
 
-  generateStyles() {
-    this.styles = createStyles(this.props);
-  }
+  // TODO: deprecate the check for object
```

Comment:
TODO hints for `isObject` check, but in `onPress` it's commented out

Reply:
I actually removed it in a following PR